### PR TITLE
tests: nrf: enable twister tests for nrf54l20pdk

### PR DESCRIPTION
**Pull request by e-rk**
Enabled the following tests on nrf54l20
- i2c_slave
- clock_control_api
- nrf_clock_calibration
- nrf_lf_clock_start
- onoff
- retained_mem/api
- uart_elementary
- uart_errors
- uart_mix_fifo_poll
- uart_pm
- watchdog
- fcb

### CODE REVIEW

File: tests/drivers/watchdog/wdt_error_cases/src/main.c [hunk 1]
```diff
@@ -42,8 +42,12 @@
 #define DEFAULT_WINDOW_MIN (0U)
 
 /* Align tests to the specific target: */
-#if defined(CONFIG_SOC_NRF54L05) || defined(CONFIG_SOC_NRF54L10) || \
```
Comment (kl-cruz) [inline tests/drivers/watchdog/wdt_error_cases/src/main.c:45]: I think nRF54L symbols can be changed to CONFIG_SOC_NRF54LX.
See:https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nordic/Kconfig.soc#L48

Comment (e-rk) [reply tests/drivers/watchdog/wdt_error_cases/src/main.c:45]: Used `CONFIG_SOC_SERIES_NRF54LX` instead.

File: tests/drivers/counter/counter_basic_api/boards/nrf54l20pdk_nrf54l20_cpuapp.overlay [hunk 2]
```diff
@@ -0,0 +1,35 @@
+&timer00 {
+	prescaler = <6>;
+	status = "okay";
+	max-frequency = <DT_FREQ_M(64)>;
```
Comment (kl-cruz) [inline tests/drivers/counter/counter_basic_api/boards/nrf54l20pdk_nrf54l20_cpuapp.overlay:4]: If neeed it should be moved to dts/common/nordic/nrf54l_05_10_15.dtsi

File: tests/drivers/uart/uart_mix_fifo_poll/boards/nrf54l20pdk_nrf54l20_cpuapp.overlay [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	current-speed = <115200>;
+	pinctrl-0 = <&uart21_default>;
+	pinctrl-1 = <&uart21_sleep>;
+	pinctrl-names = "default", "sleep";
+	hw-flow-control;
+};
+
+counter_dev: &timer00 {
+	status = "okay";
+	max-frequency = <DT_FREQ_M(64)>;
```
Comment (kl-cruz) [inline tests/drivers/uart/uart_mix_fifo_poll/boards/nrf54l20pdk_nrf54l20_cpuapp.overlay:39]: Probably to be updated in dts/common/nordic/nrf54l_05_10_15.dtsi

File: tests/drivers/counter/counter_basic_api/boards/nrf54l20pdk_nrf54l20_cpuapp.overlay (hunk 2 above)
Comment (e-rk) [reply tests/drivers/counter/counter_basic_api/boards/nrf54l20pdk_nrf54l20_cpuapp.overlay]: Moved to `nrf54l20.dtsi`

File: tests/drivers/uart/uart_mix_fifo_poll/boards/nrf54l20pdk_nrf54l20_cpuapp.overlay (hunk 3 above)
Comment (e-rk) [reply tests/drivers/uart/uart_mix_fifo_poll/boards/nrf54l20pdk_nrf54l20_cpuapp.overlay]: Moved to `nrf54l20.dtsi`

**Comment by e-rk** [comment]
@kl-cruz PR is green, please review.

**Review by masz-nordic — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**